File: aether/patch_hero_script.py
import re

with open("lib/screens/profile_screen.dart", "r") as f:
    text = f.read()

old_hero_start = text.find("  Widget _buildHeroGlassCard() {")
old_hero_end = text.find("  Widget _buildMiniSocialStat(", old_hero_start)

if old_hero_start != -1 and old_hero_end != -1:
    old_hero_full = text[old_hero_start:old_hero_end]

    with open("hero_card.dart", "r") as f:
        new_hero = f.read()
    
    new_text = text.replace(old_hero_full, new_hero)
    with open("lib/screens/profile_screen.dart", "w") as f:
        f.write(new_text)
    print("Replaced successfully")
else:
    print("Could not find bounds")

File: aether/patch_hero_script.dart
import 'dart:io';

void main() {
  final file = File('lib/screens/profile_screen.dart');
  final text = file.readAsStringSync();

  final oldHeroStart = text.indexOf('  Widget _buildHeroGlassCard() {');
  final oldHeroEnd = text.indexOf('  Widget _buildMiniSocialStat(');

  if (oldHeroStart != -1 && oldHeroEnd != -1) {
    final oldHeroFull = text.substring(oldHeroStart, oldHeroEnd);
    final newHero = File('hero_card.dart').readAsStringSync();
    
    final newText = text.replaceFirst(oldHeroFull, newHero);
    file.writeAsStringSync(newText);
    print('Replaced hero safely');
  } else {
    print('Could not find bounds');
  }
}
